core/xml.js

```diff
@@ -349,6 +349,14 @@ Blockly.Xml.domToWorkspace = function(xml, workspace) {
       }
     } else if (name == 'shadow') {
       goog.asserts.fail('Shadow block cannot be a top-level block.');
+    }  else if (name == 'variables') {
+      if (i == 1) {
```

Comment:
You are right, I did some digging found that childNodes = [text, variables, text, block] when it should be [variables, block]. The 'text' nodes show up as the first and third node. It is the indents before <variables> and <block> as seen here:

        '<xml>' +
        '  <variables>' +
        '    <variable type="type1" id="id1">name1</variable>' +
        '    <variable type="type2" id="id2">name2</variable>' +
        '    <variable type="" id="id3">name3</variable>' +
        '  </variables>' +
        '  <block type="field_variable_test_block">' +
        '    <field name="VAR" id="id3" variabletype="">name3</field>' +
        '  </block>' +
        '</xml>'

This wasn't a problem prior because we weren't checking the location of the tags